wcomponents-test-lib/src/test/java/com/github/bordertech/wcomponents/test/WDateFieldWebElementTest.java

```diff
@@ -0,0 +1,43 @@
+package com.github.bordertech.wcomponents.test;
+
+import com.github.bordertech.wcomponents.test.components.WDateFieldUI;
+import com.github.bordertech.wcomponents.test.selenium.ByButton;
+import com.github.bordertech.wcomponents.test.selenium.ByLabel;
+import com.github.bordertech.wcomponents.test.selenium.MultiBrowserRunner;
+import com.github.bordertech.wcomponents.test.selenium.SeleniumJettyTestCase;
+import com.github.bordertech.wcomponents.test.selenium.driver.SeleniumWComponentsWebDriver;
+import com.github.bordertech.wcomponents.test.selenium.element.SeleniumWButtonWebElement;
+import com.github.bordertech.wcomponents.test.selenium.element.SeleniumWDateFieldWebElement;
+import com.github.bordertech.wcomponents.test.selenium.element.SeleniumWFieldIndicatorWebElement;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MultiBrowserRunner.class)
+public class WDateFieldWebElementTest extends SeleniumJettyTestCase {
+
+
+	/**
+	 * Constructor...
+	 */
+	public WDateFieldWebElementTest() {
+		super(new WDateFieldUI());
+	}
+
+	@Test
+	public void testDateFieldUsingTextFeatures() {
+		SeleniumWComponentsWebDriver driver = getDriver();
+
+		SeleniumWButtonWebElement validatingButton = driver.findWButton(new ByButton("Validate button", false, true));
+
+		validatingButton.click();
+
+		SeleniumWDateFieldWebElement dateField = driver.findWDateField(new ByLabel("this is a datefield", false));
+
+		SeleniumWFieldIndicatorWebElement message = dateField.getFieldIndicatorMessage();
+		assertEquals(message.getText(), "this is a datefield must be completed.");
+
+		dateField.sendKeys("18102018");
```

Comment:
Have you tried running this test in a browser which supports native date fields and does this send-keys still work. I also suggest using a potentially ambiguous date such as 05062018 as the date field in some user agents is locale specific. If you are going to write functional tests which can run in a browser they have to work in all targets and be locale independent.